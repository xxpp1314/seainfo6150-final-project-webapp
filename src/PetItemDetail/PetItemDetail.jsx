import React, {useEffect, useState} from 'react'
import { isEmpty } from "lodash";
import PropTypes from "prop-types";
import styles from "./PetItemDetail.module.css";
import Form from '../Form/Form';
import PetDetailCoverImage from '../PetDetailCoverImage/PetDetailCoverImage.jsx';
import PetDetailDescription from '../PetDetailDescription/PetDetailDescription.jsx'

const PetItemDetail = (props) => {
    const [fetchedData, setFetchedData] = useState({});

    useEffect(() => {
        const fetchData = async () => {
          const result = await fetch("http://demo2087294.mockable.io/");
          const resultJson = await result.json();
          setFetchedData(resultJson);
        };
    
        if (isEmpty(fetchedData)) {
          fetchData();
        }
    }, [fetchedData]);  

    let displayContent;
    if(!isEmpty(fetchedData)) {
        var pet = fetchedData[props.category][props.categoryid];
        displayContent = (
        <div className={styles.mainpage}>
            <h1 className={styles.text}>{pet.title}</h1>
            <div className={styles.container}>
                <div className={styles.flex}>
                    <div className={styles.flexitems}>
                        <PetDetailCoverImage url={pet.img.url}/>
                    </div>
                    <div className={styles.flexitems}>
                        <PetDetailDescription detail={pet}/>
                    </div>
                </div>
            </div>
            <div className={styles.gap}></div>
            <div className={styles.form}>
                <h2 className={styles.text}>Do you want to take this pet to your home? 
                Please leave your information here.</h2>
                <Form className={styles.formdetails}/>
            </div>
        </div>
        )
    }
    else {
        displayContent = <div><center>Data is fetching</center></div>;
    }

    return (
        displayContent
    )
}

PetItemDetail.propTypes = {
    category:PropTypes.string.isRequired,
    categoryid: PropTypes.string.isRequired     
};

export default PetItemDetail;