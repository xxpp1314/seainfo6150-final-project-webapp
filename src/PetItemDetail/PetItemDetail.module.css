.mainpage{ 
    background-image: linear-gradient(to right,rgb(83, 15, 15), rgb(49, 12, 25) 70%);
    padding: 5px 10px 50px 5px;
    align-items: center;
    padding: 2%;
}

.container {
    padding: 2%;
    box-shadow: 0px 0px 5px 2px rgba(204, 0, 0, 0.5);    
    border: 0px solid grey;
    border-radius: 1%;
    display: grid;
    grid-template-rows: 1fr;
    justify-items: center;
  }

  .gap{
    padding: 1%;
  }

  .flexitems{
    padding: 1%;
    border-radius: 2%;
    width: 50%;
    box-shadow: 0px 0px 5px 2px rgba(204, 0, 0, 0.5);    
    border: 0px solid grey;
    margin: 2%;
    align-items: center;
    justify-content: center;
  }
  
  @media screen and (min-width: 100px) {
    .container {
      grid-template-rows: repeat(1, 1fr);
    }
  }
  
  @media screen and (min-width: 800px) {
    .container {
      grid-template-rows: repeat(1, 1fr);
    }
  }

  @media screen and (min-width: 100px) {
    .flexitems {
      width: 80%;
    }
  }


  @media screen and (min-width: 600px) {
    .flexitems {
      width: 80%;
      height: 100%;
    }
  }

 
  .flex{
    display:flex;
    flex-direction: row;
    height: 100%;
}

@media screen and (min-width: 100px) {
  .flex {
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
}

@media screen and (min-width: 800px) {
  .flex {
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
}

@media screen and (min-width: 1000px) {
  .flex {
    flex-direction: row;
  }
}
  
.text {
    padding: 2%;
    color:  rgb(243, 222, 222);
    text-align: center;
}

.form {
  padding-left: 10%;
  padding-right: 10%;
  padding-bottom: 2%;
  box-shadow: 0px 0px 10px 2px rgba(204, 0, 0, 0.5);    
  border: 0px solid #666;
  border-radius: 1%;
  display: grid;
  grid-template-rows: 1fr;
  justify-items: center;
}