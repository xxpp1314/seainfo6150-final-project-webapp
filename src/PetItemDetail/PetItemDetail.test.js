import React from "react";
import { render } from "@testing-library/react";
import PetItemDetail from "./PetItemDetail.jsx";

it("renders the PetItemDetail correctly", () => {
  const { container } = render(
        <PetItemDetail
            category="dogs"
            categoryid="Russell"
        />
    );
    expect(container).toMatchSnapshot();
  });