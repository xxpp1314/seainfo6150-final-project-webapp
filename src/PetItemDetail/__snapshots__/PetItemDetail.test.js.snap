// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PetItemDetail correctly 1`] = `
<div>
  <div>
    <center>
      Data is fetching
    </center>
  </div>
</div>
`;
